opencog/reasoning/RuleEngine/rule-engine-src/pln/BCPatternMatch.cc

```diff
@@ -0,0 +1,65 @@
+/*
+ * BCPatternMatch.cc
+ *
+ * Copyright (C) 2014 Misgana Bayetta
+ *
+ * Author: Misgana Bayetta <misgana.bayetta@gmail.com>  October 2014
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#include "BCPatternMatch.h"
+#include <opencog/guile/SchemeSmob.h>
+BCPatternMatch::BCPatternMatch(AtomSpace * as) :
+		Implicator(as), DefaultPatternMatchCB(as), AttentionalFocusCB(as), PLNImplicator(
+				as), as_(as) {
+
+}
+
+BCPatternMatch::~BCPatternMatch() {
+
+}
+
+bool BCPatternMatch::node_match(Handle& node1, Handle& node2) {
+	if (AttentionalFocusCB::node_match(node1, node2))
+		return false;
+	else
+		return true;
+
+}
+bool BCPatternMatch::link_match(LinkPtr& lpat, LinkPtr& lsoln) {
```

Comment:
Caution: most of these changes pre-date the recent reversal of Pattern Matcher's callback value, so likely it should be re-reversed?